### PR TITLE
feat(): Add support for combination of TaskGetItem and constant inputs

### PR DESCRIPTION
**Pull request by DominikZuercherQC**
The first implementation of the `inputs` parameter was not able to handle TaskGetItem properly leading it to be incompatible with `RawSql` tasks or tasks returning more than one output.
This is fixed in this PR.

# Checklist

- [x] Added a `docs/source/changelog.md` entry
- [ ] Added/updated documentation in `docs/source/`
- [ ] Added/updated examples in `docs/source/examples.md`

### CODE REVIEW

File: src/pydiverse/pipedag/core/flow.py [hunk 1]
```diff
@@ -395,6 +395,15 @@ def run(
         if trace_hook is None:
             trace_hook = TraceHook()
 
+        # resolve MaterializingTaskGetItem objects in inputs
+        if inputs is not None:
+            resolved_inputs = {}
+            for task, ref in inputs.items():
+                if hasattr(task, "task"):
```
Comment (DominikZuercherQC) [inline src/pydiverse/pipedag/core/flow.py:402]: Use isinstance(TaskGetItem)

Comment (DominikZuercherQC) [reply src/pydiverse/pipedag/core/flow.py:402]: This overrides all outputs of the task

File: src/pydiverse/pipedag/core/task.py [hunk 2]
```diff
@@ -204,7 +204,11 @@ def task_result_mapper(x):
             if isinstance(x, Task):
                 return x.resolve_value(inputs[x.id])
             if isinstance(x, TaskGetItem):
-                return x.resolve_value(inputs[x.task.id])
+                try:
+                    return x.resolve_value(inputs[x.task.id])
+                except TypeError:
```
Comment (windiana42) [inline src/pydiverse/pipedag/core/task.py:209]: I would prefer an instance check here. You somehow have a problem with wiring time vs. runtime types. At some point the ExternalTableReference should pop up.

**Review by windiana42 — APPROVED** [review]

File: src/pydiverse/pipedag/engine/base.py [hunk 3]
```diff
@@ -39,3 +39,33 @@ def run(
...
+    for task_identifier, reference in inputs.items():
+        task_id = task_identifier[0]
+        if task_id not in task_inputs.keys():
+            continue
+        task_item = task_identifier[1]
+        if task_item is None:
+            task_inputs[task_id] = reference
+        elif isinstance(task_item, int):
+            # handling TaskGetItem args
+            items = list(task_inputs[task_id])
```
Comment (DominikZuercherQC) [inline src/pydiverse/pipedag/engine/base.py:57]: test nout errors

File: src/pydiverse/pipedag/engine/base.py [hunk 4]
```diff
@@ -39,3 +39,33 @@ def run(
...
+            items[task_item] = reference
+            task_inputs[task_id] = tuple(items)
+        elif isinstance(task_item, str):
```
Comment (DominikZuercherQC) [inline src/pydiverse/pipedag/engine/base.py:60]: can dict key be int?

File: src/pydiverse/pipedag/engine/base.py [hunk 5]
```diff
@@ -39,3 +39,33 @@ def run(
             engine specific.
         :return: A result instance wrapping the flow execution result.
         """
+
+
+def _replace_task_inputs_with_const_inputs(
```
Comment (DominikZuercherQC) [inline src/pydiverse/pipedag/engine/base.py:44]: docstring add that caching is invalid

**Pull request merged**